gulp/tasks/preview.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Gulp tasks for managing an S3-based preview of the Pattern Library documentation.
+ *
+ * The tasks are as follows:
+ *  - build-preview: builds a preview site and uploads it to S3
+ *  - jekyll-build-preview: build the preview site
+ *  - upload-preview: upload the preview site to S3
+ *  - remove-preview: removes the preview site from S3
+ *
+ * Note: please set the environment variable S3_PREVIEW_DOMAIN to the domain
+ * used to host your S3 preview.
+ */
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        runSequence = require('run-sequence'),
+        childProcess = require('child_process'),
+        gitUtils = require('../util/gitUtils'),
+        previewConfigFile = '_tmp_preview_config.yml',
+        previewSiteDir = '_preview_site',
+        previewDomain = process.env.S3_PREVIEW_DOMAIN;
+
+    gulp.task('build-preview', function(callback) {
+        runSequence(
+            'jekyll-build-preview',
+            'upload-preview',
+            callback);
+    });
+
+    gulp.task('jekyll-build-preview', function() {
+        var branch = gitUtils.currentBranch(),
+            previewBaseUrl = '/' + branch + '/';
+        // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+        childProcess.execSync(
```

Comment:
Sorry that that wasn't clear. Each gulp task already catches errors and displays them, so having our own try/catch seems IMO verbose and unnecessary. I've looked at a lot of gulp task implementations using Google and I don't see any that do explicit error handling. As I said, gulp is more like a makefile, and in makefiles you don't write error handling.

I tried forcing some errors and while the exceptions could be clearer in some cases, the actual error is always reported. For example, here's what happens when I switch the command from 'aws' to 'awsx':

```
$ gulp upload-preview
[23:44:43] Using gulpfile ~/edx/src/ux-pattern-library/gulpfile.js
[23:44:43] Starting 'upload-preview'...
/bin/sh: awsx: command not found
[23:44:43] 'upload-preview' errored after 14 ms
[23:44:43] Error: Command failed: awsx s3 sync _preview_site s3://ux-test.edx.org/andya/s3-preview
/bin/sh: awsx: command not found

    at checkExecSyncError (child_process.js:1346:13)
    at Object.execSync (child_process.js:1386:13)
    at Gulp.<anonymous> (/Users/andya/edx/src/ux-pattern-library/gulp/tasks/preview.js:54:26)
    at module.exports (/Users/andya/edx/src/ux-pattern-library/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/andya/edx/src/ux-pattern-library/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/andya/edx/src/ux-pattern-library/node_modules/gulp/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/andya/edx/src/ux-pattern-library/node_modules/gulp/node_modules/orchestrator/index.js:134:8)
    at /usr/local/lib/node_modules/gulp-cli/lib/versioned/^3.7.0/index.js:46:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
```